tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1325,37 +1354,63 @@ static void test_provisioner_pb_remote_client_ncrp_provision(void)
 	PASS();
 }
 
+static void comp_data_get(uint16_t server_addr, uint8_t page, struct net_buf_simple *comp)
+{
+	uint8_t page_rsp;
+
+	net_buf_simple_reset(comp);
+	ASSERT_OK(bt_mesh_cfg_cli_comp_data_get(0, server_addr, page, &page_rsp, comp));
+	ASSERT_EQUAL(page, page_rsp);
+}
+
+static void comp_data_compare(struct net_buf_simple *comp1, struct net_buf_simple *comp2,
+			      bool expect_equal)
+{
+	if (expect_equal) {
+		ASSERT_EQUAL(comp1->len, comp2->len);
+		if (memcmp(comp1->data, comp2->data, comp1->len)) {
+			FAIL("Composition data is not equal");
+		}
+	} else {
+		if (comp1->len == comp2->len) {
```

Comment:
I thought about it. It does not affect this test in its current state, but technically the sizes of the two compositions could be the same without being equal?

Reply:
Alright. I just don't like ifs in the test and would like to try to avoid them. E.g. in the test we can assume that they are different (we are composing CDPs after all).

Reply:
Yes, I can see that. I just did not like the idea of someone adding something in the future using my helper function that strongly implies that it will accept different compositions with the same length.